lib/ansible/module_utils/remote_management/dellemc/dellemc_idrac.py

```diff
@@ -0,0 +1,97 @@
+# _*_ coding: utf-8 _*_
+
+#
+# Dell EMC OpenManage Ansible Modules
+# Version 1.0
+# Copyright (C) 2018 Dell Inc.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# All rights reserved. Dell, EMC, and other trademarks are trademarks of Dell Inc. or its subsidiaries.
+# Other trademarks may be trademarks of their respective owners.
+#
+
+from __future__ import (absolute_import, division,
+                        print_function)
+import tempfile
+import os
+__metaclass__ = type
+
+try:
+    from omsdk.sdkinfra import sdkinfra
+    from omsdk.sdkcreds import UserCredentials
+    from omsdk.sdkfile import FileOnShare, file_share_manager
+    from omsdk.sdkprotopref import ProtoPreference, ProtocolEnum
+    from omsdk.http.sdkwsmanbase import WsManOptions
+    HAS_OMSDK = True
+
+except ImportError:
+
+    HAS_OMSDK = False
+
+
+class iDRACConnection():
+    def __init__(self, module):
+        if HAS_OMSDK is False:
+            results = {}
+            results['msg'] = "Dell EMC OMSDK library is required for this module"
+            module.fail_json(**results)
+
+        self.module = module
+        self.handle = None
+
+    def connect(self):
+        results = {}
+
+        ansible_module_params = self.module.params
+
+        idrac = ansible_module_params.get('idrac')
+        idrac_ip = ansible_module_params.get('idrac_ip')
```

Comment:
All further args (except for `port`) are marked as required in spec so you don't need `.get()` method which suggests that they are optional.
Use item access:
```suggestion
        idrac_ip = ansible_module_params['idrac_ip']
```

This will also protect you against accidental requirement status change in future raising exception early rather than masking such bug under the pile of optionals.

Reply:
committed the suggestion for all the module params